libr/include/r_cons.h

```diff
@@ -878,8 +879,13 @@ typedef int (*RLineCallback)(RLine *line);
 
 typedef struct r_line_comp_t {
 	bool opt;
+
+	// private! use r_line_completion_argc(), r_line_completion_argv()
 	int argc;
-	const char **argv;
+	const char **argv; // weak reference to a whole list, if NULL, args will be used
+	RPVector args; /* <char *> */
+	bool args_weak; // whether args are weak refs or should be freed on clear
```

Comment:
isnt this something that should be tied to the rpvector constructor? instead of adding another var

Reply:
No, for the RPVector, weak would just mean "no free function", but I need to check this value at some places and don't really want to check if the free function is null since that would be kind of confusing.

Reply:
yes i know, but if you construct the vecvtor with NULL free callback it will behave like this, you dont need an extra var to duplicate the info already exposed in vec->free

Reply:
Yep that's what I said, but then I need to check the free function, or more precisely `vec->v.free_user`, which is kind of an internal implementation detail of RPVector. It would definitely work, but I wouldn't really like it.